src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -179,6 +234,7 @@ define(function (require, exports, module) {
 
         HealthLogger.setHealthLogsEnabled(isHDTracking);
         window.clearTimeout(timeoutVar);
+         sendAnalyticsDataToServer("pingData", "", "", "");
```

Comment:
This call is not protected in try catch. If this fails neither health data nor analytics data would be sent periodically.

Reply:
Brackets also ships with the modern `fetch` api now, so we could use that:

```js
var healthData = getAnalyticsData(category, subcategory, type, subtype);

return fetch(brackets.config.analyticsDataServerURL, {
  method: 'POST', // or 'PUT'
  body: JSON.stringify({events: [healthData]}), 
  headers: new Headers({
    'Content-Type': 'application/json',
    'x-api-key': brackets.config.projectName
  })
});

...

sendAnalyticsDataToServer("pingData", "", "", "").catch(console.err);
```

Reply:
@swmitra  This is not required  in the code,this is added just for testing